### PR TITLE
PHP 8: Code Review `WebServices`

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Hi @smeyer-ilias,

I completed the review of the `Services/WebServices` component (without the `ECS` sub-directory). In general the code looked very good in regards of the PHP types.

Results:
- [ ] The code style is `PSR-2 + X` compliant
- [x] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`
- [x] There is a `Unit Test Suite` with at least one unit test
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [x] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or explict PHP types are used (type hints, return types, typed properties)

Our goal was to eliminate the usage of super global variables. The usages in `Services/WebServices/Rest/server.php` might be okay. The `ECS` folder (not part of this review, and AFAIK not your responsibility) is still full of super global usages.

--

Furthermore the `PHP-CS-FIXER` "touched" some files because the code was not `PSR-2 + X` compliant.

I also fixed some trivial issues reported by my inspection profile.

Best regards,
@mjansenDatabay

### CODE REVIEW

File: Services/WebServices/SOAP/classes/class.ilAbstractSoapMethod.php [hunk 1]
```diff
@@ -70,11 +70,11 @@ protected function checkParameters(array $params) : void
...
      * is returned to the SOAP caller.
      * @param string $a_message
-     * @param int    $a_code
+     * @param string|int $a_code
      * @return void
-     *@throws ilSoapPluginException
+     * @throws ilSoapPluginException
      */
-    protected function __raiseError(string $a_message, $a_code)
+    protected function __raiseError(string $a_message, $a_code) : void
```
Comment (smeyer-ilias) [inline Services/WebServices/SOAP/classes/class.ilAbstractSoapMethod.php]: The parent method returns string|int
Therefore no return type here.

File: Services/WebServices/Rest/classes/class.ilRestFileStorage.php [hunk 2]
```diff
@@ -176,7 +174,7 @@ public function deleteDeprecated() : void
 
     public function writeToFile($a_data, $a_absolute_path) : bool
     {
-        if (!$fp = fopen($a_absolute_path, 'w+')) {
+        if (!$fp = fopen($a_absolute_path, 'wb+')) {
```
Comment (smeyer-ilias) [inline Services/WebServices/Rest/classes/class.ilRestFileStorage.php:177]: For interest: I thought binary mode is default and therefore not required. Is this assumption wrong?

**Review by smeyer-ilias — CHANGES_REQUESTED** [review]
Thanks for reviewing.
There is one issue regarding a wrong return type.

File: Services/WebServices/SOAP/classes/class.ilAbstractSoapMethod.php (hunk 1 above)
Comment (mjansenDatabay) [reply Services/WebServices/SOAP/classes/class.ilAbstractSoapMethod.php]: Okay, the parent method returns ` soap_fault|SoapFault|null`. I'll fix this.

**Review by smeyer-ilias — APPROVED** [review]
Thank for reviewing ant the improvements.

**Pull request merged**